windows/RNGestureHandler/RNGestureHandler/RNGestureHandler.h

```diff
@@ -0,0 +1,147 @@
+#pragma once
+
+#include "pch.h"
+#include "NativeModules.h"
+#include "RNGestureRegistry.h"
+
+using namespace winrt::Microsoft::ReactNative;
+
+namespace winrt::RNGestureHandler
+{
+    // In sync with react-native-gesture-handler/src/State.ts
+    enum class RNGestureHandlerState
+    {
+        Undetermined = 0,
+        Failed = 1,
+        Began = 2,
+        Cancelled = 3,
+        Active = 4,
+        End = 5,
+    };
+
+    struct RNGestureHandlerEventExtraData
+    {
+        RNGestureHandlerEventExtraData();
+
+        // Pan-specific
+        float translationX;
+        float translationY;
+        float velocityX;
+        float velocityY;
+
+        // Rotation/Pinch-specific
+        float rotation;
+        float velocity;
+        float anchorX;
+        float anchorY;
+
+        // LongPress/Pan/Tap-specific
+        float x;
+        float y;
+        float absoluteX;
+        float absoluteY;
+
+        // Pinch-specific
+        float scale;
+        float focalX;
+        float focalY;
+    };
```

Comment:
Shouldn't we have different structs for different extra data?

Reply:
We could but it introduces more complexity in the rest of the code and doesn't really gain much.